pkg/reconcilermanager/controllers/otel_sa_controller.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"kpt.dev/configsync/pkg/metrics"
+	"kpt.dev/configsync/pkg/status"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ reconcile.Reconciler = &OtelSAReconciler{}
+
+const defaultSAName = "default"
+
+// OtelSAReconciler reconciles the default service account under the config-management-monitoring namespace.
+type OtelSAReconciler struct {
+	clusterName string
+	client      client.Client
+	log         logr.Logger
+	scheme      *runtime.Scheme
+}
+
+// NewOtelSAReconciler returns a new OtelSAReconciler.
+func NewOtelSAReconciler(clusterName string, client client.Client, log logr.Logger, scheme *runtime.Scheme) *OtelSAReconciler {
+	if clusterName == "" {
+		clusterName = "unknown_cluster"
```

Comment:
Should we error out if no cluster is specified?

Reply:
Interesting that it's not used but still specified, could be a pattern copied from the r*sync reconcilers. We might remove it as it's not used in here, unless we want to include cluster name in the log

Reply:
Let us keep it to be consistent with [kpt-config-sync/pkg/reconcilermanager/controllers/otel_controller.go](https://github.com/GoogleContainerTools/kpt-config-sync/blob/688b7a1d7c57e3f8b5167b34d318125ec7c782f5/pkg/reconcilermanager/controllers/otel_controller.go#L55).